lib/ruby_lexer.rb

```diff
@@ -1177,7 +1181,10 @@ def parse_string quote # TODO: rewrite / remove
       return :tSTRING_END, nil
     end
 
-    space = true if qwords and scan(/\s+/)
+    if qwords and scan(/\s+/)
+      space = true
+      self.extra_lineno += matched.count("\n")
```

Comment:
I used the `extra_lineno` as a place to store the newlines. I originally tried directly updating `self.lineno`, but it seems that no matter what I did, I would bump the line right before a token was parsed and an s-expression was made (since whitespace is the delimiter for when we are done parsing a token). So, instead of directly bumping the line number I store it here and then bump it when we create the s-expression later.